frontend/packages/dev-console/src/components/pipelines/PipelineVisualizationGraph.scss

```diff
@@ -36,30 +62,109 @@ $gutter: 1.7em;
           height: 0;
         }
       }
+      & .odc-pipeline-vis-task {
+        &__connector {
+          display: none;
+          &::after {
+            content: '';
+            width: 0.67em;
+            height: 1.2em;
+            position: absolute;
+            top: -2.75em;
+            border-top: 1px solid $border-color;
+            left: -17px;
+            border-right: 1px solid var(--pf-global--BorderColor--light-100);
+            border-radius: 0 20px;
+          }
+        }
+        &:not(:first-child) {
+          //connect each task
+          &::before {
+            content: '';
+            border-radius: 0 0 0 1.33em;
+            top: -4.33em;
+            position: absolute;
+            border-bottom: 1px solid var(--pf-global--BorderColor--light-100);
+            width: 10px;
+            height: 5.75em;
+          }
+          // Left connecting lines
+          &::before {
+            left: -10px;
+            border-left: 1px solid $border-color;
+            border-radius: 0 0 0 12px;
+          }
+        }
+        &:nth-child(2) {
+          .odc-pipeline-vis-task__connector {
+            display: block;
+          }
+          &::before {
+            top: -1.9em;
+            height: 3.3em;
+          }
+        }
+      }
     }
-  }
-
-  &__stage.is-parallel {
-    & .odc-pipeline-vis-task {
-      &:first-child {
-        &::before {
-          content: '';
-          position: absolute;
-          top: 1.25em;
-          border-top: 1px solid $border-color;
-          width: $gutter;
-          height: 0;
+    &:not(:last-child) {
+      &.is-parallel {
+        & .odc-pipeline-vis-task {
+          &:first-child {
+            &::after {
+              right: 0;
+              transform: translateX(100%);
+            }
+          }
         }
-        &::before {
-          left: 0;
-          transform: translateX(-100%);
+      }
+      & .odc-pipeline-vis-task {
+        &__connector {
+          display: none;
+          &::before {
+            content: '';
+            width: 0.67em;
+            height: 1.2em;
+            position: absolute;
+            top: -2.75em;
+            border-top: 1px solid $border-color;
+            right: -17px;
+            border-left: 1px solid var(--pf-global--BorderColor--light-100);
+            border-radius: 20px 0 0;
+          }
         }
-        &::after {
-          right: 0;
-          transform: translateX(100%);
+        //connect each task
+        &:not(:first-child) {
```

Comment:
Next three blocks of code are duplicated on top of the file as well, we could avoid this using `extends`